app/controllers/matchings_controller.rb

```diff
@@ -0,0 +1,224 @@
+# frozen_string_literal: true
+
+class MatchingsController < ApplicationController
+  before_action :set_instance_variables, only: [:show, :edit, :update, :destroy, :email, :modify]
+
+  def set_show_variables
+    @proj_exists = !(@project.nil?)
+    if @proj_exists
+      @permission = current_user.id == @project.user.id
+    end
+
+    if @proj_exists and @matching
+      @parsed_matching = JSON.parse(@matching.output_json)
+
+    elsif @proj_exists
+      @all_submitted_preferences = all_submitted_preferences?
+    end
+
+    respond_to do |format|
+      format.html
+      format.csv {send_data Matching.to_csv(@matching.output_json), :filename => @project.project_name + "_matching.csv"}
+    end
+  end
+
+  def modify
+    @matching = Matching.find_by(params.slice(:project_id))
+    @parsed_matching = JSON.parse(@matching.output_json)
+    !@parsed_matching["schedule"][params[:format].split('/')[0].to_i]["timestamp"]  = params[:format].split('/')[1]
+    !@matching.output_json = @parsed_matching.to_json
+    @matching.save
+    redirect_to project_matching_path(params[:project_id])
+  end
+
+  # GET /project/:project_id/matching
+  def show
+    unless logged_in?
+      require_user
+      return
+    end
+
+    set_show_variables
+  end
+
+  def email
+    @project = Project.find(params[:project_id])
+    @participants = @project.participants
+    @email_subject = params[:email_subject]
+    @email_body = params[:email_body]
+    @parsed_matching = JSON.parse(@matching.output_json)
+    ParticipantsMailer.set_project_name(@project.project_name)
+
+    emails_to_times = {}
+    @parsed_matching['schedule'].each do |matching|
+      email = matching['people_called'][0]
+      timestamp = Time.parse(matching["timestamp"]).strftime("%A, %B %d %Y %I:%M %p")
+
+      if !emails_to_times[email]
+        emails_to_times[email] = ""
+      end
+      emails_to_times[email] += timestamp + ", "
+    end
+
+    emails_to_times.each do |email, times|
+      ParticipantsMailer.matching_email(email, @email_subject, @email_body, times[0...-2]).deliver_now
+    end
+
+    flash[:success] = 'Emails have been sent.'
+    redirect_to project_matching_path(params[:project_id])
+  end
+
+  # GET /projects/:project_id/matching/new
+  # def new
+  #   @matching = Matching.new
+  # end
+
+  # TODO: convert to a PUT/ UPDATE
+  # GET /projects/:project_id/matching/edit
+
+  # def edit
+  #   respond_to do |format|
+  #     if all_submitted_preferences?
+  #       #matching = Matching.where(project_id: @project.id).update_all(output_json: api)
+  #       if @matching.update(output_json: api)
+  #         flash[:success] = 'Successfully matched.'
+  #         format.html { redirect_to project_matching_path }
+  #       # else
+  #       #   format.html { render :edit }
+  #       end
+  #     end
+  #   end
+  # end
+
+  # POST /projects/:project_id/matching
+  def create
```

Comment:
Please put most of the logic into the model instead of cthe controller

Reply:
Addressed in [this commit](https://github.com/saasbook/MultiMeet/pull/17/commits/34f23ba6d1051169f3d52cf51e36f8eec3688648)